packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,416 @@
+import type { AnonCredsCredentialValue } from '@aries-framework/anoncreds'
+import type { Agent, FileSystem, WalletConfig } from '@aries-framework/core'
+import type { EntryObject } from '@hyperledger/aries-askar-shared'
+
+import { AnonCredsCredentialRecord, AnonCredsLinkSecretRecord } from '@aries-framework/anoncreds'
+import { AskarWallet } from '@aries-framework/askar'
+import { InjectionSymbols, KeyDerivationMethod, JsonTransformer, TypedArrayEncoder } from '@aries-framework/core'
+import { Migration, Key, KeyAlgs, Store, StoreKeyMethod } from '@hyperledger/aries-askar-shared'
+
+import { IndySdkToAskarMigrationError } from './errors/IndySdkToAskarMigrationError'
+import { transformFromRecordTagValues } from './utils'
+
+/**
+ *
+ * Migration class to move a wallet form the indy-sdk structure to the new
+ * askar wallet structure.
+ *
+ * Right now, this is ONLY supported within React Native environments AND only sqlite.
+ *
+ * The reason it only works within React Native is that we ONLY update the
+ * keys, masterSecret and credentials for now. If you have an agent in Node.JS
+ * where it only contains these records, it may be used but we cannot
+ * guarantee a successful migration.
+ *
+ */
+export class IndySdkToAskarMigrationUpdater {
+  private store?: Store
+  private walletConfig: WalletConfig
+  private defaultLinkSecretId: string
+  private agent: Agent
+  private dbPath: string
+  private fs: FileSystem
+  private deleteOnFinish: boolean
+
+  private constructor(
+    walletConfig: WalletConfig,
+    agent: Agent,
+    dbPath: string,
+    deleteOnFinish = false,
+    defaultLinkSecretId?: string
+  ) {
+    this.walletConfig = walletConfig
+    this.dbPath = dbPath
+    this.agent = agent
+    this.fs = this.agent.dependencyManager.resolve<FileSystem>(InjectionSymbols.FileSystem)
+    this.defaultLinkSecretId = defaultLinkSecretId ?? walletConfig.id
+    this.deleteOnFinish = deleteOnFinish
+  }
+
+  public static async initialize({
+    dbPath,
+    agent,
+    deleteOnFinish,
+    defaultLinkSecretId,
+  }: {
+    dbPath: string
+    agent: Agent
+    deleteOnFinish?: boolean
+    defaultLinkSecretId?: string
+  }) {
+    const {
+      config: { walletConfig },
+    } = agent
+    if (typeof process?.versions?.node !== 'undefined') {
+      agent.config.logger.warn(
+        'Node.JS is not fully supported. Using this will likely leave the wallet in a half-migrated state'
+      )
```

Comment:
I'm not happy with this. A half migrated wallet is not worth a thing. We should either not support Node.JS or add support for the missing properties or throw an error when we detect that the wallet has credential definition record upfront.

Reply:
Not needed before merging

Reply:
The only reason I added this is because we have nodejs tests. I can do a hack thing by overriding the global `process.version` check we do as well. 

I agree that this is suboptimal.

Reply:
Overriding `process.versions.node` does not seem to work as straight forward as I thought. I could check if its a Jest environment inside the `initialise` code, but that does not seem too clean.

Reply:
Detecting credential definition might actually be the best solution. I will add that, I assume we do not have to check other fields as well.

Reply:
Yeah that's a good one. Will make everybody of us half-happy!.